services/horizon/internal/flags.go

```diff
@@ -133,6 +133,24 @@ func Flags() (*Config, support.ConfigOptions) {
 			Usage:       "HTTP port for Captive Core to listen on (0 disables the HTTP server)",
 			ConfigKey:   &config.CaptiveCoreHTTPPort,
 		},
+		&support.ConfigOption{
+			Name:        "captive-core-storage-path",
+			OptType:     types.String,
+			FlagDefault: "",
+			CustomSetValue: func(opt *support.ConfigOption) {
+				existingValue := *opt.ConfigKey.(*string)
```

Comment:
I can be wrong but I think the `ConfigKey` isn't set at this point (can you check). In the other places I can see we are using `viper.Get*` method so in this case it should be:
```suggestion
				existingValue := viper.GetString(co.Name)
```

Reply:
Oh interesting; I could've sworn I modeled this directly after the other uses of `CustomSetValue` but I guess not exactly.

Reply:
Fixed in 32e72c6de9b1ea5b26a4b60792328a46d82b0938!